shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -302,18 +303,30 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
             typer.dbg = mode.debugVariance
             typer.computeVariances(curBlockTypeDefs, ctx)
             typer.dbg = temp
-            
+            val varianceWarnings: MutSortedMap[Str, Ls[Str]] = MutSortedMap()
+
             // process type definitions first
             typeDefs.foreach(td =>
               // check if type def is not previously defined
               if (ctx.tyDefs.contains(td.nme.name)
                   && !oldCtx.tyDefs.contains(td.nme.name)) {
                   // ^ it may not end up being defined if there's an error
+
                 val tn = td.nme.name
                 val ttd = ctx.tyDefs(tn)
+
+                // generate warnings for bivariant type variables
+                val bivariantTypeVars = ttd.tparamsargs.filter{ case (tname, tvar) =>
+                  ttd.tvarVariances.get(tvar).map(v => v.equals(typer.VarianceInfo.bi)).getOrElse(false)
```

Comment:
Can it happen in a well-typed program that the variance not be found in `ttd.tvarVariances`? If yes, under what circumstances? => a comment explaining this and justifying the `.getOrElse(false)` is needed.

Also, don't use Java's `.equals`. Just use `===` (it's shadowed by ScalaTest here, so use the `=:=` synonym).

Also, the current logic could be written:
```suggestion
                  ttd.tvarVariances.get(tvar).contains(typer.VarianceInfo.bi)
```

Reply:
There really is no situation when variance will not be found. Inside computeVariances we are initializing the variance for all type parameters to bi. The refactor you've shared looks like the best way to express this.